src/Compilers/Core/Portable/PEWriter/ManagedResource.cs

```diff
@@ -40,7 +42,7 @@ public void WriteData(BlobBuilder resourceWriter)
             {
                 try
                 {
-                    using (Stream stream = _streamProvider())
+                    using (Stream stream = _streamProvider!())
```

Comment:
Consider adding an assert instead of using a suppression.

Reply:
➡️ see https://github.com/dotnet/roslyn/pull/39053#discussion_r331666865

Reply:
This one can absolutely be directly verified with an Assert here. We _will_ null ref if `_streamProvider` is null.

Reply:
I took to preferring the exclusions because the preferable approach would be finding a compiler-verifiable way to write this code (i.e. no assertion or suppression is needed). Either way, it can be addressed in a small follow-up pull request.